Backend/app/services/db_service.py

```
@@ -0,0 +1,85 @@
+    audience = audience_resp.data[0]
+
+    # Fetch all sponsorships
+    sponsorships_resp = supabase.table("sponsorships").select("*").execute()
+    matches = []
+    for sponsorship in sponsorships_resp.data:
+        match_score = 0
+        # Audience age group overlap
+        if 'required_audience' in sponsorship and 'audience_age_group' in audience:
+            required_ages = sponsorship['required_audience'].get('age_group', [])
+            creator_ages = audience.get('audience_age_group', {})
+            overlap = sum([creator_ages.get(age, 0) for age in required_ages])
```

Pylint (3.3.7)

[refactor] Consider using a generator instead 'sum(creator_ages.get(age, 0) for age in required_ages)'

(R1728)